core/sourcehook/sh_memory.h

```diff
@@ -328,12 +328,6 @@ namespace SourceHook
 
 			prevHandler = signal(SIGSEGV, BadReadHandler);
 
-			volatile const char *p = reinterpret_cast<const char*>(addr);
```

Comment:
Without this, the function doesn't do anything (here it's trying to potentially read from a bad address in order to trigger a `SIGSEGV`). What was the compiler complaining about?

Reply:
char dummy below being unused, `p` also looked unused to me outside of the loop just under the declaration. what do you suggest as a way to improve this section if we can’t remove it

Reply:
Just remove `dummy` but put `p` and the loop back, just accessing `p` is enough and the compiler can't optimise it away.